src/main/java/buchen/station/CitiBikeController.java

```diff
@@ -0,0 +1,197 @@
+
+
+        track.clear();
+        track.add(startLocation);
+        track.add(new GeoPosition(startStation.lat, startStation.lon));
+        track.add(new GeoPosition(endStation.lat, endStation.lon));
+        track.add(endLocation);
+
+
+        RoutePainter routePainter = new RoutePainter(track);
+
+        List<Painter<JXMapViewer>> painters = List.of(
+                waypointPainter,
+                routePainter
+        );
+
+        CompoundPainter<JXMapViewer> painter = new CompoundPainter<>(painters);
+        mapViewer.setOverlayPainter(painter);
+
+        mapViewer.zoomToBestFit(
+                Set.of(
+                        startLocation,
+                        new GeoPosition(startStation.lat, startStation.lon),
+                        new GeoPosition(endStation.lat, endStation.lon),
+                        endLocation
+                ),
+                1.0
+        );
+        mapViewer.repaint();
+    }
+
+    public void setStartPoint(BiConsumer<Double, Double> listener) {
+        this.startPointDouble = listener;
+    }
+
+    public void setEndPoint(BiConsumer<Double, Double> listener) {
+        this.endPointDouble = listener;
+    }
+
+    private void updateWaypoints() {
+        WaypointPainter<Waypoint> waypointPainter = new WaypointPainter<>();
+        Set<Waypoint> waypoints = new HashSet<>();
+
+        if (startLocation != null) {
+            waypoints.add(new DefaultWaypoint(startLocation));
+        }
+        if (endLocation != null) {
+            waypoints.add(new DefaultWaypoint(endLocation));
+        }
+        waypointPainter.setWaypoints(waypoints);
+        mapViewer.setOverlayPainter(waypointPainter);
+        mapViewer.repaint();
+    }
+
+    public void clear() {
+        start = false;
+        startLocation = null;
+        endLocation = null;
+        mapViewer.setOverlayPainter(null);
+        mapViewer.repaint();
+    }
+
+    public CitiBikeRequest writeToJson() {
+        Point from = new Point();
+        from.lat = startLocation.getLatitude();
+        from.lon = startLocation.getLongitude();
+
+        Point to = new Point();
+        to.lat = endLocation.getLatitude();
+        to.lon = endLocation.getLongitude();
+
+        CitiBikeRequest request = new CitiBikeRequest();
+        request.from = from;
+        request.to = to;
+
+        Gson gson = new GsonBuilder().setPrettyPrinting().create();
+        try (FileWriter writer = new FileWriter("request.json")) {
+            gson.toJson(request, writer);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        return request;
+    }
+
+    public CitiBikeResponse getLambda(CitiBikeRequest request) {
+        CitiBikeResponse citibikeResponse = null;
+        LambdaService api = new LambdaServiceFactory().getService();
+
+        try {
+            citibikeResponse = api.getClosestStations(request).blockingGet();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
If an exception does occur, this just causes a null pointer exception when you return null from this method.